filesystem.go

```diff
@@ -0,0 +1,59 @@
+package mmark
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// FileSystem implements access to files. The elements in a file path are
+// separated by slash ('/', U+002F) characters, regardless of host
+// operating system convention.
+type FileSystem interface {

```

Comment:
Think just having a path is more end-user friendly, considering the exported interface is mostly a C interface (blackfriday is 1:1 copy of some C code). Put an interface type out there, is, I think, maybe a step too far.